src/native/libs/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -30,11 +43,8 @@ if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
     append_extra_compression_libs(NATIVE_LIBS_EXTRA)
 
     if (CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
-        include(${CLR_SRC_NATIVE_DIR}/external/zlib.cmake)
-        add_definitions(-DINTERNAL_ZLIB)
-        set_source_files_properties(${ZLIB_SOURCES} PROPERTIES COMPILE_DEFINITIONS "${ZLIB_COMPILE_DEFINITIONS}")
-        set_source_files_properties(${ZLIB_SOURCES} PROPERTIES COMPILE_OPTIONS "${ZLIB_COMPILE_OPTIONS}")
-        set(NATIVECOMPRESSION_SOURCES ${ZLIB_SOURCES} ${NATIVECOMPRESSION_SOURCES})
+        set(NATIVECOMPRESSION_SOURCES ${NATIVECOMPRESSION_SOURCES})
+        list(APPEND NATIVE_LIBS_EXTRA zlib)
```

Comment:
We noticed that this might need to be added on all targets, not just Browser and Wasi - or at least determine what zlib gets passed to the linker below on https://github.com/dotnet/runtime/pull/102403/files#diff-5cd57a2173307aaec58099abe5dc12175d1e5aa72a35c2a2b283a735d8af158cR74